pkg/trace/api/telemetry.go

```diff
@@ -0,0 +1,207 @@
+// on best effort basis. In order to duplicate the data across many instances of DD.
+func (pec *proxyEndpointsConfig) buildAdditionalTargets() (targets []target) {
+	additionalEndpoitnsCfg := fmt.Sprintf("apm_config.%s.additional_endpoints", pec.name)
+
+	if config.Datadog.IsSet(additionalEndpoitnsCfg) {
+		extra := config.Datadog.GetStringMapStringSlice(additionalEndpoitnsCfg)
+		for endpoint, keys := range extra {
+			u, err := url.Parse(endpoint)
+			if err != nil {
+				log.Errorf("Error parsing additional %s intake URL %s: %v", pec.name, endpoint, err)
+				continue
+			}
+			for _, key := range keys {
+				targets = append(targets, target{
+					url:    u,
+					apiKey: key,
+				})
+			}
+		}
+	}
+	return targets
+}
+
+// telemetryProxyHandler parses returns a new HTTP handler which will proxy requests to the configured intakes.
+// If the main intake URL can not be computed because of config, the returned handler will always
+// return http.StatusInternalServerError along with a clarification.
+func (r *HTTPReceiver) telemetryProxyHandler() http.Handler {
+	if !telemetryProxyEndpointsConfig.isEnabled() {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			msg := fmt.Sprintf("Telemetry proxy forwarder is Disabled")
+			http.Error(w, msg, http.StatusMethodNotAllowed)
+		})
+	}
+	baseTarget, err := telemetryProxyEndpointsConfig.buildBaseTarget(r.conf.APIKey())
+
+	if err != nil {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			msg := fmt.Sprintf("Telemetry proxy forwarder encountered error: %v", err)
+			http.Error(w, msg, http.StatusInternalServerError)
+		})
+	}
+
+	transport := telemetryMultiTransport{
+		rt:                r.conf.NewHTTPTransport(),
+		baseTarget:        *baseTarget,
+		additionalTargets: telemetryProxyEndpointsConfig.buildAdditionalTargets(),
+		director: func(req *http.Request) {
+			req.Header.Set("DD-Agent-Hostname", r.conf.Hostname)
+			req.Header.Set("DD-Agent-Env", r.conf.DefaultEnv)
+		},
+	}
+
+	limitedLogger := logutil.NewThrottled(5, 10*time.Second) // limit to 5 messages every 10 seconds
+	logger := stdlog.New(limitedLogger, "telemetry.Proxy: ", 0)
+
+	return newReverseProxy(&transport, logger)
+}
+
+// telemetryMultiTransport sends HTTP requests to multiple targets using an
+// underlying http.RoundTripper. API keys are set separately for each target.
+// The target hostname
+// When multiple endpoints are in use the response from the main endpoint
+// is proxied back to the client, while for all aditional endpoints the
+// response is discarded. There is no de-duplication done between endpoint
+// hosts or api keys.
+//
+// can be extended in the future to allow supporting more product endpoints
+// by simply parametrizing metric tags, and logger names
+type telemetryMultiTransport struct {
+	rt                http.RoundTripper
+	baseTarget        target
+	additionalTargets []target
```

Comment:
I don't think we need a new `target` type. Can we just use the same approach as the other `multiTransport` here: https://github.com/DataDog/datadog-agent/blob/eb6a11858279d402982d93bfc00bfeb32072b3cc/pkg/trace/api/profiles.go#L133-L137 ?

Also, is there an opportunity here to re-use the other `multiTransport`?

Reply:
I'd have to refactor it a bit. Because the list of target ursl and list of target keys, is IMO inferior to a list of target structs  and an explicit main struct.